src/main/kotlin/org/koil/image/Models.kt

```diff
@@ -0,0 +1,25 @@
+package org.koil.image
+
+import org.springframework.data.annotation.Id
+import org.springframework.data.annotation.LastModifiedDate
+import org.springframework.data.relational.core.mapping.Table
+import java.time.Instant
+import java.util.*
+
+@Table("profile_image")
+data class ProfileImage(
+    @Id val id: Long?,
+    val publicId: UUID,
+    val accountId: Long,
+    @LastModifiedDate val updatedAt: Instant
+) {
+    companion object {
+        fun createForAccount(accountId: Long): ProfileImage =
+            ProfileImage(
+                id = null,
```

Comment:
I find myself wishing that we created objects liked this that didn't have a nullable Id. I know this would mean we need a separate object for the creation - and I don't know if it's even possible with the repository pattern. But I've noticed already today I'm spending a lot of time doing a nullability check on the id when retrieving it from the db. It's a little clunky. 

@Table("profile_image")
data class ProfileImageCreate(@Id val id: Long?, ...)

@Table("profile_image")
data class ProfileImage(@Id val id: Long, ...)

Otherwise I'm not minding the db setup so far!